qadence/backends/pulser/backend.py

```diff
@@ -351,27 +351,14 @@ def expectation(
                     noise=noise,
                 )
                 support = sorted(list(circuit.abstract.register.support))
-                # TODO: There should be a better check for batched density matrices.
-                if dms.size()[0] > 1:
-                    res_list = [
-                        [
-                            obs.native(
-                                dm.squeeze(), param_values, qubit_support=support, noise=noise
-                            )
-                            for dm in dms
-                        ]
-                        for obs in observable
-                    ]
-                    res = torch.stack(
-                        [torch.transpose(torch.stack(res), 0, -1) for res in res_list]
-                    )
-
-                else:
-                    res_list = [
-                        obs.native(dms, param_values, qubit_support=support) for obs in observable
+                res_list = [
+                    [
+                        obs.native(dm.squeeze(), param_values, qubit_support=support, noise=noise)
+                        for dm in dms
                     ]
```

Comment:
Isn't this removing some functionality related to batching?

Reply:
From the previous code, it did not seem batching was allowed since we were looping over `dms` and `observable`. But let me check this by in the appropriate test

Reply:
We have two tests in tests/backends/pulser_basic/test_quantum_pulser where we test a non-batched case with one noise_probs value called `test_noisy_simulations`, and another case where we test several values called `test_batched_noisy_simulations`.